### PR TITLE
Container build refactor

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: packages/common/src/task/build-container.ts [hunk 1]
```diff
@@ -40,63 +37,43 @@ export class BuildContainerTask extends Task<ContainerImage> {
...
-			buildargs: {
-				PROVIDER: this.provider,
-			},
-			t: this.container.getImageTagName(this.provider),
-			dockerfile,
-		};
+		const imageId = this.container.getImageTagName(this.provider);
+		const { args = {} } = this.container.getDescriptor();
+		// Create a temporary default ignore file
+		// and delete it when we're done
```
Comment (jyecusch) [inline packages/common/src/task/build-container.ts]: Is this comment leftover from old code, I don't see an ignore file being created here.

File: packages/common/src/task/build-container.ts [hunk 2]
```diff
@@ -40,63 +37,43 @@ export class BuildContainerTask extends Task<ContainerImage> {
...
 				},
-			);
-		});
+			});
+
+			// Only outputs on stderr
+			dockerProcess.stderr.on('data', (data) => {
+				// fs.writeFileSync('debug.txt', data);
+				this.update(data.toString());
+			});
```
Comment (jyecusch) [inline packages/common/src/task/build-container.ts:64]: Does the build output get displayed or logged anywhere (beside stderr)? Thinking about cases with large base images, for example, seeing download progress would be pretty useful. Could be a separate story to add good quality output without too much noise during the build.

**Review by jyecusch — CHANGES_REQUESTED** [review]
Looking good, just a couple of comments/questions.

**Review by davemooreuws — APPROVED** [review]

File: packages/common/src/task/build-container.ts (hunk 2 above)
Comment (tjholm) [reply packages/common/src/task/build-container.ts:64]: Docker build using buildkit only outputs on stderr see: https://github.com/moby/buildkit/issues/1186.

Stdout is reserved for build artifact output.

It looks like there are options we can use to modify the buildkit progress output though. Will take a look at what we can do in a new issue.

File: packages/common/src/task/build-container.ts (hunk 1 above)
Comment (tjholm) [reply packages/common/src/task/build-container.ts]: removed.

File: packages/common/src/task/build-container.ts (hunk 2 above)
Comment (tjholm) [reply packages/common/src/task/build-container.ts:64]: Have put in --progress plain which cleans up the output somewhat, there is probably more we can do to make it cleaner though.

**Review by jyecusch — APPROVED** [review]

**Pull request merged**